mlflow/langchain/__init__.py

```diff
@@ -343,9 +351,9 @@ def _save_model(model, path):
     model_data_path = os.path.join(path, _MODEL_DATA_FILE_NAME)
     model_data_kwargs = {_MODEL_DATA_KEY: _MODEL_DATA_FILE_NAME}
 
-    if type(model) == langchain.chains.llm.LLMChain:
+    if isinstance(model, langchain.chains.llm.LLMChain):
         model.save(model_data_path)
-    elif type(model) == langchain.agents.agent.AgentExecutor:
+    elif isinstance(model, langchain.agents.agent.AgentExecutor):
```

Comment:
@sunishsheth2009 Would all subclasses of AgentExecutor work out-of-the-box? Would you prefer I revert this line of change? Thanks!

Reply:
Besides, I did not see a test for spark_udf with AgentExecutor. Does it mean we don't support this use case?

Reply:
I verified that AgentExecutor works well in spark_udf.